src/screening/bloodSugar/BloodSugarTestForm.jsx

```diff
@@ -5,39 +5,36 @@ import { Grid, Row, FormGroup, ControlLabel, Col } from 'react-bootstrap';
 import { ENCOUNTER_TYPES, CONCEPTS, FORM_ANSWERS} from "../../constants";
 import '../../assets/css/tabs.css';
 import ScreeningForm from "../ScreeningForm";
-import { formValueSelector } from "redux-form";
+import {change, formValueSelector, untouch} from "redux-form";
 import { noPaddingLeftAndRight, flexBaseline } from "../../pwaStyles";
 
 
 class BloodSugarTestForm extends React.PureComponent {
 
+  componentDidUpdate(prevProps) {
+    if (typeof this.props.fsTestType.value !== 'undefined' &&
+      typeof prevProps.fsTestType.value !== 'undefined' &&
+      this.props.fsTestType.value !== prevProps.fsTestType.value) {
+      this.clearField(this.props.bloodGlucose.fieldName);
```

Comment:
Annoying, to get the validations to work, I had to clear the out the glucose field when you changed the type... hopefully this is acceptable.